internal/timeperiod/timeperiod.go

```diff
@@ -88,11 +135,11 @@ func (e *Entry) Contains(t time.Time) bool {
 		log.Printf("Can't initialize entry: %s", err)
 	}
 
-	if t.Before(e.Start) {
+	if t.Before(e.StartTime.Time()) {
```

Comment:
Seeing as you are already touching some code here, can you please adjust the existing error handling here and probably in some other receivers as well, and always rely on the `Entry` being initialised and panic otherwise. That should then make it possible to drop all `log` package usages.